### PR TITLE
http server shutdown fix-ups [DEVINFRA-959]

### PR DESCRIPTION
**Pull request by silverjam**
from #375 
> Having some trouble shutting down the server when running in k8s. This prevents two ctrl-c handlers from being set up while in server mode (not sure this is the issue) and cleans up the termination code a bit (definitely not the issue)

implemented graceful shutdown SIGTERM handler shown here: https://stackoverflow.com/a/71279547/749342

### CODE REVIEW

File: crates/esthri/src/ops/upload.rs [hunk 1]
```diff
@@ -529,7 +529,11 @@ fn upload_part_requests<'a>(
 ) -> impl Stream<Item = UploadPartRequest> + 'a {
     let last_part = {
         let remaining = file_size % part_size;
-        (remaining > 0).then(|| (file_size / part_size + 1, remaining))
+        if remaining > 0 {
```
Comment (silverjam) [inline crates/esthri/src/ops/upload.rs:532]: Required for MSRV of Rust 1.56.1

File: crates/esthri-internals/Cargo.toml [hunk 2]
```diff
@@ -9,11 +9,13 @@ repository = "https://github.com/swift-nav/esthri/"
 homepage = "https://github.com/swift-nav/esthri/"
 readme = "../../README.md"
 keywords = ["aws", "s3"]
+rust-version = "1.56.1"
```
Comment (silverjam) [inline crates/esthri-internals/Cargo.toml:12]: If we want to support earlier rust versions, we'd need to remove this key

**Review by silverjam — COMMENTED** [review]
I used the tool `cargo-msrv` to validate that 1.56.1 is in fact the lowest version we support: https://crates.io/crates/cargo-msrv

File: Makefile.toml [hunk 3]
```diff
@@ -7,22 +7,23 @@ VERBOSE = ""
 TLS_PROVIDER = "rustls"
 TARGET = ""
 WORKSPACE = "${CARGO_MAKE_WORKING_DIRECTORY}"
+MSRV = ""
 
 [env.release]
 BUILD_FLAVOR = "--release"
-RUSTFLAGS="-C link-arg=-fuse-ld=lld -C codegen-units=1 -C panic=abort -C lto=fat -C opt-level=3"
```
Comment (gborough) [inline Makefile.toml:13]: Maybe in the future we can experiment with using the Mold linker if we really care about linking performance

**Comment by notoriaga** [comment]
> I used the tool `cargo-msrv` to validate that 1.56.1 is in fact the lowest version we support: https://crates.io/crates/cargo-msrv

very cool!

File: Makefile.toml (hunk 3 above)
Comment (notoriaga) [reply Makefile.toml:13]: I've been using it for a while, makes a big difference

**Review by notoriaga — APPROVED** [review]

File: crates/esthri/Cargo.toml [hunk 4]
```diff
@@ -9,6 +9,7 @@ repository = "https://github.com/swift-nav/esthri/"
 homepage = "https://github.com/swift-nav/esthri/"
 readme = "../../README.md"
 keywords = ["aws", "s3"]
+rust-version = "1.56.1"
```
Comment (gborough) [inline crates/esthri/Cargo.toml:12]: we can also create a .clippy.toml file at the project root dir that specifies msrv = "1.56.1"

**Review by gborough — APPROVED** [review]

File: crates/esthri/Cargo.toml (hunk 4 above)
Comment (silverjam) [reply crates/esthri/Cargo.toml:12]: Cool, feel free to submit a PR for this :-)

**Pull request merged**